src/bot.ts

```diff
@@ -227,6 +227,9 @@ export class DiscordBot {
 
         client.on("messageDelete", async (msg: Discord.Message) => {
             try {
+                if (await this.store.isGuildChannelBanned(msg.guild.id, msg.channel.id)) {
```

Comment:
go only over channel id, `msg.guild` may even be undefined.

the channel ID is already a 100% unique way to identify all channels no matter what

Reply:
So the reason we use guild id in the db is for readability as much as anything. It's important to preserve the guild => channel mapping for that purpose. I guess we could search by either channel or guild tho.

Reply:
this will severely break the bot once we add something for non-guild channels down the road, though (e.g. command interface via DM?) due to those not having a guild.